patches/net/minecraft/util/ExtraCodecs.java.patch

```diff
@@ -0,0 +1,32 @@
+--- a/net/minecraft/util/ExtraCodecs.java
++++ b/net/minecraft/util/ExtraCodecs.java
+@@ -558,6 +_,7 @@
+ 
+    static class RecursiveCodec<T> implements Codec<T> {
+       private final Supplier<Codec<T>> wrapped;
++      private final java.util.concurrent.atomic.AtomicInteger recursion = new java.util.concurrent.atomic.AtomicInteger();
```

Comment:
This might be better off being a thread-local boolean, so that way multiple threads trying to call toString concurrently don't get unpredictable results.

Reply:
We discussed this in Discord and yeah, a synchronized method is likely better than a threadlocal or atomic integer.